go/base_test.go

```diff
@@ -0,0 +1,49 @@
+package testGoSDK
+
+import (
+	"flag"
+	"log"
+	"os"
+	"testing"
+
+	xenapi "github.com/xenserver/xenserver-samples/go/goSDK"
+)
+
+var IP_FLAG = flag.String("ip", "", "the URL of the form https://ip[:port] pointing at the server")
+var USERNAME_FLAG = flag.String("username", "", "the username of the host (e.g. root)")
+var PASSWORD_FLAG = flag.String("password", "", "the password of the host")
+var CA_CERT_PATH_FLAG = flag.String("ca_cert_path", "", "the CA certificate file path for the host")
+var NFS_SERVER_FLAG = flag.String("nfs_server", "", "the ip address pointing at the nfs server")
+var NFS_PATH_FLAG = flag.String("nfs_path", "", "the nfs server path")
+var IP1_FLAG = flag.String("ip1", "", "the URL of the form https://ip[:port] pointing at another host1")
+var USERNAME1_FLAG = flag.String("username1", "", "the username of the host1 (e.g. root)")
+var PASSWORD1_FLAG = flag.String("password1", "", "the password of the host1")
+
+var session *xenapi.Session
+
+func TestLogin(t *testing.T) {
+	session = xenapi.NewSession(&xenapi.ClientOpts{
+		URL: "http://" + *IP_FLAG,
+		Headers: map[string]string{
+			"User-Agent": "XS Go SDK v24.3.0",
```

Comment:
I think the User-Agent should characterise the client, i.e. these tests, not the SDK version used (this can change depending on the module we download). I suggest using something like "XS SDK for Go - Examples v1.0".